### PR TITLE
refactor: Reorganize variable ownership

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/context/code_scope.h [hunk 1]
```diff
@@ -15,18 +15,43 @@
...
 
 namespace hlasm_plugin::parser_library::context {
 class location_counter;
 class set_symbol_base;
 
+struct code_scope_variable
+{
+    set_symbol_base* ref = nullptr;
+
+    std::variant<std::monostate, set_symbol<A_t>, set_symbol<B_t>, set_symbol<C_t>> data;
```
Comment (jirimosinger) [inline parser_library/src/context/code_scope.h:32]: Maybe we could have a using for this? Something like
`using set_symbol_variant = std::variant<std::monostate, set_symbol<A_t>, set_symbol<B_t>, set_symbol<C_t>>`

Not sure where to put it though. Maybe right here?

File: parser_library/src/debugging/debugger.cpp [hunk 2]
```diff
@@ -387,11 +387,11 @@ class debugger::impl final : public processing::statement_analyzer
 
         for (const auto& [_, value_type] : proc_stack_[frame_id].scope.variables)
         {
-            const auto& [value, global] = value_type;
+            const auto& [ref, _data, global] = value_type;
```
Comment (jirimosinger) [inline parser_library/src/debugging/debugger.cpp:390]: Consider using `__` instead
```suggestion
            const auto& [ref, __, global] = value_type;
```

**Review by jirimosinger — APPROVED** [review]

File: parser_library/src/context/code_scope.h (hunk 1 above)
Comment (slavek-kucera) [reply parser_library/src/context/code_scope.h:32]: I am not sure if it adds anything. The type should not really be used in the rest of the code.

File: parser_library/src/debugging/debugger.cpp (hunk 2 above)
Comment (slavek-kucera) [reply parser_library/src/debugging/debugger.cpp:390]: I had that and then decided to change it back :)

**Pull request merged**